scripts/pylint_extensions.py

```diff
@@ -961,6 +961,57 @@ def process_module(self, node):
                 self.add_message('single-space-after-yield', line=line_num + 1)
 
 
+class ExcessiveEmptyLinesChecker(checkers.BaseChecker):
+    """Checks if there are excessive newlines between function definations."""
+    __implements__ = interfaces.IRawChecker
+
+    name = 'excessive-new-lines'
+    priority = -1
+    msgs = {
+        'C0011': (
+            'Excessive new lines between function definations.',
+            'excessive-new-lines',
+            'Remove extra newlines.'
+        )
+    }
+
+    def process_module(self, node):
+        """Process a module.
```

Comment:
This doc is lacking in detail, explain what is being done here in more detail.

Reply:
Done.